unlock-app/src/actions/user.ts

```diff
@@ -62,3 +66,12 @@ export const gotPassword = (password: string) => ({
   type: GOT_PASSWORD,
   password,
 })
+
+// This should be dispatched along with setAccount in the storage middleware
+// when a user logs in. This provides a bit of a timing concern, since
+// SET_ACCOUNT usually wipes out all the state. So SET_ACCOUNT must be sent
+// first.
+export const setEncryptedPrivateKey = (key: EncryptedPrivateKey) => ({
```

Comment:
Take note of the comment above. Not sure what the best way to enforce this is.

Reply:
What if you also passed the account along with the key? This way you can index them by account, and if the account changes to the account using the key, we don't blow it away?

Reply:
I see, so we’d use SET_ACCOUNT for crypto users, and this one for those with accounts? I’m amenable to that, though it may require some code changes in consumers of the Redux store

Reply:
No I mean we pass in the account to this action. But, more to the point, do we need to reset this when account changes? If we get a password, that means we only have one possibility, our account. So maybe we just remove the reset?